lib/exceptions.ts

```diff
@@ -1,55 +1,107 @@
+type Message = string;
+
+interface Status {
+  status: number;
+  statusText: string;
+}
+
+interface ErrorDetails {
+  signature?: string;
+  raw?: any;
+}
+
+interface FetchErrorDetails extends Status {
+  headers: Headers;
+  body: string;
+}
+
+// Deprecated
+interface AxiosErrorDetails extends Partial<Status> {
+  originalError: Error;
+  code?: string;
+}
+
 export class SignatureValidationFailed extends Error {
-  constructor(
-    message: string,
-    public signature?: string,
-  ) {
+  public signature?: string;
+
+  constructor(message: Message, { signature }: ErrorDetails = {}) {
     super(message);
+    this.name = this.constructor.name;
+
+    Object.assign(this, { signature });
   }
 }
 
 export class JSONParseError extends Error {
+  public raw: any;
+
+  constructor(message: Message, { raw }: ErrorDetails = {}) {
+    super(message);
+    this.name = this.constructor.name;
+
+    Object.assign(this, { raw });
+  }
+}
+
+export class HTTPFetchError extends Error {
```

Comment:
Unless there is a special reason, for the sake of simplifying the differences, I would like you to move `HTTPFetchError` back to its original position (at the bottom).

Reply:
`RequestError`, `ReadError`, and `HTTPError` are error classes that are currently used only in http-axios.ts (deprecated) files. All three classes were deprecated with Axios and moved to /* Deprecated */ bottom of the annotation.
```
1. Move back to original position.
2. Retain current position (the three recoded classes are located at the bottom of the annotation)
```
There may be a reason for the code I haven't checked, so please check and give me feedback! :)